### ISSUE TITLE
Timeouts in instrumentation crash the instrumented process

### ISSUE THREAD

**Issue by lukaszsamson** [appsignal/appsignal-elixir#354]
version: 1.6.3
In our production app we have some GenServers with decorated callbacks like
```
defmodule MyApp.MyProcess do
use GenServer
@decorate transaction(:background_job)
 def handle_info(msg, state) do
    ...
 end
end
```

Current behavior:
Timeouting instrumentation crashes our processes

```
Process :"Elixir.MyApp.MyProcess" (#PID<0.2648.0>) terminating: {:timeout, {GenServer, :call, [Appsignal.TransactionRegistry, {:monitor, #PID<0.2648.0>}, 5000]}}

(elixir) lib/gen_server.ex:836: GenServer.call/3
(appsignal) lib/appsignal/transaction/registry.ex:39: Appsignal.TransactionRegistry.register/1
(appsignal) lib/appsignal/transaction.ex:84: Appsignal.Transaction.register/1
(afi) lib/my_app/my_process.ex:1: MyApp.MyProcess.handle_info/2
(stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
(stdlib) gen_server.erl:686: :gen_server.handle_msg/6
(stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

```
GenServer :"Elixir.MyApp.MyProcess" terminating
** (stop) exited in: GenServer.call(Appsignal.TransactionRegistry, {:remove, AppSignal.Transaction{9an9jpl5fed5g}}, 5000)
** (EXIT) time out
(elixir) lib/gen_server.ex:836: GenServer.call/3
(appsignal) lib/appsignal/transaction.ex:375: Appsignal.Transaction.complete/1
(afi) lib/my_app/my_process.ex:1: MyApp.MyProcess.handle_info/2
(stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
(stdlib) gen_server.erl:686: :gen_server.handle_msg/6
(stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

```
Process MyApp.MyProcess (#PID<0.14976.2>) terminating
** (exit) exited in: GenServer.call(Appsignal.TransactionRegistry, {:monitor, #PID<0.14976.2>}, 5000)
** (EXIT) time out
```

Expected behavior:
Instrumentation is not critical. A failure should be isolated from the instrumented code (error kernel pattern). An error should be logged.
Also I can see that the instrumentation code is synchronous. This does not seem right.

**Comment by jeffkreeftmeijer** [comment]
Hey @lukaszsamson 👋 

You're right. This shouldn't happen. Thanks for this detailed explanation. We're investigating this, and we'll keep you posted on our progress.

**Comment by tombruijn** [comment]
Linked PR was closed #356. Closing issue
